### PR TITLE
ref(k8s): tidy up Python logic in k8s scheduler

### PR DESCRIPTION
**Pull request by helgi**
Consolidate code and unify a bit more under single quote (as much of the code already is).
Rename raise_err to error to be clearer about its functionality

Clean up logic in _delete_pod and have waiting work in it.

More to come, split out to simplify review

### CODE REVIEW

**Comment by bacongobbler** [comment]
most of it is just switching around names and adding whitespace, but the chained `replace` calls and `unhealthy` function calls are a nice cleanup. +1

**Comment by helgi** [comment]
The reason for the whitespaces was to have control structures separated out instead having them smashed together in one long function.  Def a personal preference on that one but nothing essential :)

`_delete_pod` needs the closest look since the logic in it looked broken so I used my best judgement to fix it.

**Comment by bacongobbler** [comment]
nope. The logic for both is the exact same AFAICT. It just looks awkward with the way it was set up. Your implementation is more clear as to its intention.

File: rootfs/scheduler/k8s.py [hunk 1]
```diff
@@ -145,8 +153,8 @@ def _get_old_rc(self, name, app_type):
                 break
         if exists:
             return prev_rc
-        else:
```
Comment (mboersma) [inline rootfs/scheduler/k8s.py]: You don't like the explicit `if` -- `else`? I've been writing a lot of Go code so this feels right, but not sure which is more pythonic.

Ok, I've already spent too much time typing about this detail. :+1:

Comment (helgi) [inline rootfs/scheduler/k8s.py]: The way I've always approached it (can't remember if it is in K&R or similar style guides) is that you are already returning a default value at the end, no need to `else` those, `exists` is the special condition, etc etc :)

**Pull request merged**